src/bilder/components/hashicorp/consul/models.py

```diff
@@ -101,9 +101,9 @@ class Consul(HashicorpProduct):
     def systemd_template_context(self):
         return self
 
-    def render_configuration_files(self) -> Iterable[Tuple[Path, str]]:
+    def render_configuration_files(self) -> Iterable[Tuple[str, str]]:
         for fpath, config in self.configuration.items():  # noqa: WPS526
-            yield self.configuration_directory.joinpath(fpath), config.json(
+            yield str(self.configuration_directory.joinpath(fpath)), config.json(
```

Comment:
Rather than doing these string coercions I've got PyInfra pinned to 1.4.4 which still works with Path objects as inputs. We should stick with that until the project adds support for Paths in a more holistic manner.